### PR TITLE
Revert "Use an explicit Agent (TMB Client) to send messages to Foreman."

### PR DESCRIPTION
**Pull request by zuyu**
This reverts commit be14790b0ced59cb7c6f802464ba7337eb7fbe77.

Instead, will use an explicit scheduler client id for receiving TMB messages.

### CODE REVIEW

File: relational_operators/CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +91,7 @@ target_link_libraries(quickstep_relationaloperators_BuildHashOperator
 target_link_libraries(quickstep_relationaloperators_CreateIndexOperator
                       glog
                       quickstep_catalog_CatalogRelation
-                      quickstep_relationaloperators_RelationalOperator
+                      quickstep_relationaloperators_RelationalOperator
```
Comment (hbdeshmukh) [inline relational_operators/CMakeLists.txt]: Can you get rid of the trailing spaces here?

**Comment by hbdeshmukh** [comment]
Looks good to me. Will merge after Travis finishes.

File: relational_operators/CMakeLists.txt (hunk 1 above)
Comment (zuyu) [inline relational_operators/CMakeLists.txt]: No, not in this one. This PR just only does the reverse.

**Comment by zuyu** [comment]
@hbdeshmukh There are block counting issues in the CI.

**Pull request merged**